gensrc/proto/lake_types.proto

```diff
@@ -70,6 +70,11 @@ message DelfileWithRowsetId {
     optional string name = 1;
     // origin rowset that generate this del file.
     optional uint32 origin_rowset_id = 2;
+    // If there is a upsert and delete mix transaction, we use
+    // `op_offset` to identify the order of deletes.
+    // E.g. there are <upsert, upsert, upsert, delete> in a transaction,
+    //      so the `op_offset` of this delete is 2.
```

Comment:
what does this `op_offset` mean, don't understand why the order is 2 in a `upsert,upsert,upsert,delete` batch.

Reply:
DELETE don't have its own rssid, so its rssid will be same as its previous UPSERT.

Reply:
so it is a 0-based index, and shares the same index as the previous record. that makes it [0, 1, 2, 2]. I was thinking [1,1,1,2]. That's where the misunderstanding comes.

Reply:
Yes, I will make it more clear in comment.

Reply:
done